ts/WoltLabSuite/Core/Acp/Ui/User/Editor.ts

```diff
@@ -64,80 +66,34 @@ class AcpUiUserEditor {
-          data: {
-            actionName: "com.woltlab.wcf.user.sendNewPassword",
-            parameters: {
-              confirmMessage: Language.get("wcf.acp.user.action.sendNewPassword.confirmMessage"),
-              objectIDs: [userId],
-            },
-          },
-          responseData: {
-            actionName: "com.woltlab.wcf.user.sendNewPassword",
-            objectIDs: [userId],
-          },
-        });
-      });
-    }
-
     const deleteContent = dropdownMenu.querySelector(".jsDeleteContent") as HTMLAnchorElement;
     if (deleteContent !== null) {
       new AcpUserContentRemoveHandler(deleteContent, userId);
     }
 
+    const sendNewPassword = dropdownMenu.querySelector(".jsSendNewPassword") as HTMLAnchorElement;
+    if (sendNewPassword !== null) {
+      new SendNewPasswordAction(sendNewPassword, userId, userRow);
+    }
+
     const toggleConfirmEmail = dropdownMenu.querySelector(".jsConfirmEmailToggle") as HTMLAnchorElement;
     if (toggleConfirmEmail !== null) {
-      toggleConfirmEmail.addEventListener("click", (event) => {
-        event.preventDefault();
+      new ToggleConfirmEmailAction(toggleConfirmEmail, userId, userRow);
+    }
 
-        Ajax.api(
-          {
-            _ajaxSetup: () => {
-              const isEmailConfirmed = Core.stringToBool(userRow.dataset.emailConfirmed!);
-
-              return {
-                data: {
-                  actionName: (isEmailConfirmed ? "un" : "") + "confirmEmail",
-                  className: "wcf\\data\\user\\UserAction",
-                  objectIDs: [userId],
-                },
-              };
-            },
-          } as AjaxCallbackObject,
-          undefined,
-          (data: DatabaseObjectActionResponse) => {
-            document.querySelectorAll(".jsUserRow").forEach((userRow: HTMLTableRowElement) => {
-              const userId = ~~userRow.dataset.objectId!;
-              if (data.objectIDs.includes(userId)) {
-                const confirmEmailButton = dropdownMenu.querySelector(".jsConfirmEmailToggle") as HTMLAnchorElement;
-
-                switch (data.actionName) {
-                  case "confirmEmail":
-                    userRow.dataset.emailConfirmed = "true";
-                    confirmEmailButton.textContent = confirmEmailButton.dataset.unconfirmEmailMessage!;
-                    break;
-
-                  case "unconfirmEmail":
-                    userRow.dataset.emailEonfirmed = "false";
-                    confirmEmailButton.textContent = confirmEmailButton.dataset.confirmEmailMessage!;
-                    break;
-
-                  default:
-                    throw new Error("Unreachable");
-                }
-              }
-            });
-
-            UiNotification.show();
-          },
-        );
-      });
+    const enableUser = dropdownMenu.querySelector(".jsEnable") as HTMLAnchorElement;
+    if (enableUser !== null) {
```

Comment:
These types don't match up. Doesn't tsc warn you about this? In the first line you are already asserting that `enableUser` is an `HTMLAnchorElement`, this implies that it can never be null. You must either remove the check for `!== null` or use `as HTMLAnchorElement|null` (or whatever the syntax is).